Source/WebCore/platform/graphics/avfoundation/objc/CDMSessionAVContentKeySession.mm

```diff
@@ -376,8 +385,22 @@ static bool isEqual(Uint8Array* data, const char* literal)
     return Uint8Array::tryCreate(static_cast<const uint8_t*>([m_expiredSession bytes]), [m_expiredSession length]);
 }
 
+bool CDMSessionAVContentKeySession::hasContentKeyRequest() const
+{
+    LockHolder holder { m_keyRequestLock };
+    return m_keyRequest;
+}
+
+RetainPtr<AVContentKeyRequest> CDMSessionAVContentKeySession::contentKeyRequest()
+{
+    LockHolder holder { m_keyRequestLock };
+    return RetainPtr { m_keyRequest.get() };
+}
+
 void CDMSessionAVContentKeySession::didProvideContentKeyRequest(AVContentKeyRequest *keyRequest)
 {
+    LockHolder holder { m_keyRequestLock };
+    m_hasKeyRequestSemaphore.signal();
     m_keyRequest = keyRequest;
```

Comment:
Shouldn't you signal the semaphore after setting `m_keyRequest`?

Reply:
Oh, good point!

Reply:
(It won't matter in practice because the `m_keyRequestLock` will protect access to the variable till after this method returns, but you're right that it's currently incorrect.)